dev/deps/spark-deps-hadoop-2.6

```diff
@@ -138,7 +138,7 @@ lz4-java-1.4.0.jar
 machinist_2.11-0.6.1.jar
 macro-compat_2.11-1.1.1.jar
 mail-1.4.7.jar
-mesos-1.3.0-shaded-protobuf.jar
+mesos-1.4.0-shaded-protobuf.jar
```

Comment:
@skonto Was there a specific reason for upgrading to 1.4?

Reply:
Yes I think having the latest is a good think on the master branch as long as we dont break things. 
Also this issue shows to me that several things are getting deprecated and should move on.
Any reason against it?

Reply:
OK, makes sense.

Reply:
The docs say that Spark requires >=1.0 IIRC, so as long as it still _works_ with 1.0+ (i.e. we're still sending valid messages) I think it's fine. Maybe we should update the docs with any limitations given older versions (assuming there are any)?

Reply:
@ArtRand 1.4 is compatible with 1.0+ right?